File: Creational patterns/singleton/python_model/code/LoggerFactory.py
from ConsoleLogger import ConsoleLogger
from FileLogger import FileLogger

class LoggerFactory:
    def isFileLoggingEnabled(self):
        try:
            p = __import__('.fileproperties', fromlist=["FileLogging"])
            fileLogginValue = p.FileLogging
            if fileLogginValue == "ON": return True
            else: return False
        except ModuleNotFoundError as detail:
            return False

    #Factory method
    def getLogger(self):
        if self.isFileLoggingEnabled():
            return FileLogger()
        else:
            return ConsoleLogger()

File: CreationalPatterns/factory_method/exercise/java_model/code/FactoryMethod/src/LoggerFactory.java
import java.io.*;
import java.util.*;

public class LoggerFactory {

  public boolean isFileLoggingEnabled() {
    Properties p = new Properties();
    try {
      p.load(ClassLoader.getSystemResourceAsStream("Logger.properties"));
      String fileLoggingValue = p.getProperty("FileLogging");
      if (fileLoggingValue.equalsIgnoreCase("ON") == true) return true;
      else return false;
    } catch (IOException e) {
      return false;
    }
  }

  // Factory Method
  public Logger getLogger() {
    if (isFileLoggingEnabled()) {
      return new FileLogger();
    } else {
      return new ConsoleLogger();
    }
  }
}
